### ISSUE TITLE
Feature: add assertions

### ISSUE THREAD

**Issue by vitiral** [malloydata/malloy#1290]
### What happens?

I'm new to the language, so please take this with a grain of salt. But even doing an initial "demo" project I found myself wanting to have some kind of `assert` functionality to ensure that as I hacked the data I wasn't accidentally dropping columns.

IMO this should be put inside of `run` and would use something like [implicit sources](https://malloydata.github.io/documentation/language/query#implicit-sources). This would have the added benefit of an error message being able to show the input data that caused the failure.

```
run: somethingWithPercents -> {
  project: myPercent is somePerc + otherPerc, -- whoops, that's not right!
  -- assertions expect an EMPTY result
  assert: "expect percentages to be [0-100]" is -> {
    where: not myPercent ? 0 to 100
  }
}
```

Especially nice is if the assert framework could reverse all of the final `where`/`having` clauses in the final query block and display _that_ data with the failure message -- something which _feels_ very doable although perhaps I'm not thinking of every possible case.

### To Reproduce

N/A

### OS:

N/A

### Malloy Client:

N/A

### Malloy Client Version:

N/A

### Database Connection:

N/A

**Comment by lloydtabb** [comment]
Yes, there have been a number of requests for testing.  For the moment we'd prefer that you handle it with convention.

```
source: foo is ... {
  measure: myPercent is somePerc + otherPerc, -- whoops, that's not right!
  query: my_query is {
     group_by: something
     aggregate: myPercent
   }
  
  query: assert_my_query is -> my_query refine { // should return no rows
     having: 
       myPercent > 100
       or somethingelse
   }
 }
 ```

**Comment by lloydtabb** [comment]
That said,  I do really like your idea about building assertions into the query itself...

**Comment by vitiral** [comment]
I didn't know about `refine`! I'm fairly confused about how it causes an assertion/failure to trigger though (?)

**Comment by lloydtabb** [comment]
You would keep a list of queries to run and use our command line tool to
run them.  If any of the queries returned any data, it would be a failure.
